libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/TripServiceImpl.kt

```diff
@@ -0,0 +1,145 @@
+import com.mapbox.navigation.base.trip.TripNotification
+import com.mapbox.navigation.base.trip.TripService
+import com.mapbox.navigation.navigator.MapboxNativeNavigatorImpl
+import com.mapbox.navigation.utils.extensions.ifNonNull
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import timber.log.Timber
+
+const val NAVIGATION_SERVICE_ID = 0xff00aa
+const val UPDATE_INTERVAL_IN_MILLISECONDS: Long = 1000
+const val FASTEST_UPDATE_INTERVAL_IN_MILLISECONDS: Long = 500
+
+@ExperimentalCoroutinesApi
+internal object NavigationNotificationService : Service() {
+    private val job = SupervisorJob()
+    val serviceScope = CoroutineScope(job + Dispatchers.IO)
+
+    var locationEngineCallback = object : LocationEngineCallback<LocationEngineResult> {
+        override fun onSuccess(result: LocationEngineResult?) {
+        }
+
+        override fun onFailure(exception: java.lang.Exception) {
+        }
+    }
+    var locationEngine: LocationEngine? = null
+    val locationEngineRequest: LocationEngineRequest = LocationEngineRequest.Builder(UPDATE_INTERVAL_IN_MILLISECONDS)
+            .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY)
+            .setFastestInterval(FASTEST_UPDATE_INTERVAL_IN_MILLISECONDS)
+            .build()
+
+    override fun onBind(intent: Intent?): IBinder? {
+        return null
+    }
+
+    override fun onCreate() {
+        startForeground(NAVIGATION_SERVICE_ID, null)
+        notificationChannel.close()
+        notificationChannel = ConflatedBroadcastChannel()
+        serviceScope.launch {
+            while (isActive) {
+                val status = MapboxNativeNavigatorImpl.getStatus(java.util.Date()) // Request that will drive the Polling functionality
+                locationEngine?.getLastLocation(object : LocationEngineCallback<LocationEngineResult> {
+                    override fun onSuccess(result: LocationEngineResult?) {
+                        ifNonNull(result, result?.lastLocation,
+                                { _, rawLocation ->
+                                    notificationChannel.offer(NavigationTripDescriptor(NAVIGATION_SERVICE_ID,
+                                            status.routeProgress, rawLocation, status.enhancedLocation))
+                                })
+                    }
+
+                    override fun onFailure(exception: Exception) {
+                    }
+                })
+                delay(UPDATE_INTERVAL_IN_MILLISECONDS) // Poll every second
+            }
+        }
+        locationEngine?.requestLocationUpdates(locationEngineRequest, locationEngineCallback, null)
```

Comment:
As discussed, raw fixes coming from the location engine should be handled in `TripSession` or another entity consumed by `TripSession`. Not sure though, if taking the location updates receiver out of the service would have any implications i.e. not sure if location updates will be received from trip session at all times even if the app goes to the background, the system enters in Doze mode, or anything else. Supposedly, if the service is running, the app won't be killed, but not sure if the system might affect other operations running out of the service (I've seen many unexpected behaviors working with service and long running operations in the past 😅). So ideally, it'd be great if the service is only in charge of the notification (as we're creating a foreground service) but we need to test.

Reply:
We're passing `null` as `Looper`. Any implications of doing that? Passing `null` means `invoke callbacks on the main thread` and we're consuming last location updates inside a coroutine. Are we good?

Reply:
`LastLocationEngine` has been removed. Now all we are doing is starting the notification from the service. Hence passing `null` should be just fine as the `notification` is shown on the `Main Thread`